fastpool/src/bounded.rs

```diff
@@ -195,6 +195,58 @@ impl<M: ManageObject> Pool<M> {
         })
     }
 
+    /// Replenishes the pool with at most `most` number of new objects.
+    ///
+    /// Returns `n` where `n` is the number of objects added to the pool.
+    pub async fn replenish(&self, most: usize) -> usize {
+        let permit = {
+            let mut n = most;
+            loop {
+                match self.permits.try_acquire(most) {
+                    Some(permit) => break permit,
+                    None => {
+                        n = n.min(self.permits.available_permits());
+                        continue;
+                    }
+                }
+            }
+        };
+
+        if permit.permits() == 0 {
+            return 0;
+        }
+
+        let (gap, permit) = {
+            let mut permit = permit;
+            let idle_count = self.slots.lock().deque.len();
+            if idle_count >= permit.permits() {
+                return 0;
+            }
+
+            permit.release(idle_count);
+            (permit.permits(), permit)
+        };
+
+        let mut idles = vec![];
+        for _ in 0..gap {
+            if let Ok(o) = self.manager.create().await {
+                let status = ObjectStatus::default();
+                let state = ObjectState { o, status };
+                idles.push(state);
+            }
+        }
+        let replenished = idles.len();
```

Comment:
Maybe schedule in concurrent, like using https://github.com/yoshuawuyts/futures-concurrency

But this would introduce quite a lot of complexity. Maybe we shouldn't continue this patch at all.

Reply:
No. We can release permit for every object created.